### DISCUSSION TITLE
Branching Strategy: Should I create my PR for main or for a specific RFC?

### DISCUSSION THREAD

**Discussion by thomas-k-cameron** [smithy-lang/smithy-rs#2174] [Q&A] [answered]
I'm trying to create a PR for RFC30 (serialization and de-serialization).  

I want to create several small PRs instead of a one big one. Should I create a PR for main or for some other branch?  
If so, which branch?

related: https://github.com/awslabs/smithy-rs/pull/1944#issuecomment-1371160038

**Comment by Velfi** [comment, marked as answer]
Please target PRs to `main`. Since your feature won't break anything if it's implemented piece-by-piece, that'll be fine. If you add Rust code that you don't want customers to start interacting with, consider marking it as `#[doc(hidden)]` until it's ready.
